csd/kyuubi-csd-src/descriptor/service.sdl

```diff
@@ -0,0 +1,157 @@
+      },
+      "logging": {
+	"dir" : "/var/log/kyuubi",
+        "filename" : "kyuubi-server-${host}.log",
+        "modifiable" : true,
+        "configName" : "conf/log4j2.properties",
+        "loggingType" : "log4j",
+        "additionalConfigs":[
+] 
+      },
+      "parameters": [
+        {
+          "name": "kyuubi.frontend.bind.port",
+          "label": "kyuubi  bind  port",
+          "configName": "kyuubi.frontend.bind.port",
+          "description": "Kyuubi jdbc connection port ",
+          "required": "true",
+          "type": "port",
+          "default": 10009
+        },
+		{
+          "name": "kyuubi.engine.share.level",
+          "label": "kyuubi.engine.share.level",
+          "configName": "kyuubi.engine.share.level",
+          "description": "Engines will be shared in different levels",
+          "required": "true",
+		  "default": "USER",
+		  "type": "string_enum",
+		  "validValues": ["CONNECTION","USER","GROUP","SERVER"]
+          
+        },
+        {
+          "name": "kyuubi.session.work.path",
+          "label": "kyuubi.session.work.path",
+          "configName": "kyuubi.session.work.path",
+          "description": "Engines will be shared in different levels",
+          "required": "true",
+		  "type": "path",
+		  "pathType": "localDataDir",
+          "default": "/data/kyuubi/work"
+        },
+		{
+          "name": "kyuubi.session.engine.idle.timeout",
+          "label": "Doris Frontend query Port",
+          "configName": "kyuubi.session.engine.idle.timeout",
+          "description": "engine timeout, the engine will self-terminate when it's not accessed for this duration. 0 or negative means not to self-terminate.",
+          "required": "true",
+          "type": "string",
+          "default": "PT10H"
+        }
+      ],
+      "sslServer": {
+        "keyIdentifier": "kyuubiserver",
+        "enabledConfigName": "kyuubi.ssl",
+        "keystorePasswordCredentialProviderCompatible": false,
+        "keystorePasswordScriptBased": false,
+        "keystoreKeyPasswordScriptBased": false,
+        "keystoreKeyPasswordCredentialProviderCompatible": false,
+        "keyPasswordOptionality": "required"
+      },
+      "configWriter": {
+        "generators": [
+          {
+            "filename": "conf/kyuubi-defaults.conf",
+            "configFormat": "properties",
+            "includedParams": [
+	     	    "kyuubi.frontend.bind.port",
+            	"kyuubi.engine.share.level",
+            	"kyuubi.session.engine.idle.timeout"
+            ]
+          },
+		  {
+            "filename": "conf/kyuubi-env.sh",
+            "configFormat": "properties",
+            "includedParams": []
+          }
+        ]
+      }
+    }	
+  ],
+  "gateway" : {
+    "alternatives" : {
+      "name" : "conf",
+      "priority" : 51,
+      "linkRoot" : "/etc/kyuubi"
+    },
+    "configWriter" : {
+       "peerConfigGenerators" : [
+          {
+            "filename": "conf/server.properties",
```

Comment:
what's this?

Reply:
It is used to generate client configuration, such as hive's hive-site.xml, which is saved on the gateway machine. The main reason for adding here is that there will be corresponding prompts to modify the configuration on the cm page. If you delete it, you will not be prompted. If you have any good suggestions, save some information in it